### PR TITLE
Fix epydoc warnings in twisted/words

### PR DESCRIPTION
**Pull request by rodrigc**
See:
http://twistedmatrix.com/trac/ticket/8536

### CODE REVIEW

File: twisted/words/protocols/jabber/client.py [hunk 1]
```diff
@@ -160,15 +160,15 @@ class BasicAuthenticator(xmlstream.ConnectAuthenticator):
     U{JEP-0077<http://www.jabber.org/jeps/jep-0077.html>}.
 
     Under normal circumstances, the BasicAuthenticator generates the
-    L{xmlstream.STREAM_AUTHD_EVENT} once the stream has authenticated. However,
+    C{xmlstream.STREAM_AUTHD_EVENT} once the stream has authenticated. However,
```
Comment (adiroiban) [inline twisted/words/protocols/jabber/client.py]: since this is public API I think that we should expose the documentation for STREAM_AUTHD_EVENT and link to it

File: twisted/words/protocols/irc.py [hunk 2]
```diff
@@ -3604,7 +3604,7 @@ def assembleFormattedText(formatted):
 
     @rtype: C{str}
     @return: String containing mIRC control sequences that mimic those
-        specified by L{formatted}.
+        specified by C{formatted}.
```
Comment (adiroiban) [inline twisted/words/protocols/irc.py]: minor comment... maybe use I{formatted} as C{formatted} should be used for code example and this is in fact a reference to the formatted argument

File: twisted/words/im/interfaces.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def logOn(chatui):
 
         @type chatui: Implementor of C{IChatUI}
 
-        @rtype: L{Deferred} L{Client}
+        @rtype: L{Deferred} L{Client<IClient>}
```
Comment (adiroiban) [inline twisted/words/im/interfaces.py]: why not L{IClient} ?

File: twisted/words/protocols/jabber/ijabber.py [hunk 4]
```diff
@@ -41,9 +41,10 @@ class IIQResponseTracker(Interface):
     naturally with deferreds. You send out a request and when the response
     comes back a deferred is fired.
 
-    The L{IQ} class implements a C{send} method that returns a deferred. This
-    deferred is put in a dictionary that is kept in an L{XmlStream} object,
-    keyed by the request stanzas C{id} attribute.
+    The L{<twisted.words.protocols.jabber.client.IQ>} class implements a C{send}
```
Comment (acabhishek942) [inline twisted/words/protocols/jabber/ijabber.py]: this should probably be changed as no text is written for the link. http://epydoc.sourceforge.net/epytext.html see section 3.3

File: twisted/words/protocols/jabber/client.py [hunk 5]
```diff
@@ -85,6 +85,14 @@ class IQAuthInitializer(object):
     This protocol is defined in
     U{JEP-0078<http://www.jabber.org/jeps/jep-0078.html>} and mainly serves for
     compatibility with pre-XMPP-1.0 server implementations.
+
+    @cvar INVALID_USER_EVENT: Token to signat that authentication failed, due
```
Comment (ralphm) [inline twisted/words/protocols/jabber/client.py]: Typo: `signat`

File: twisted/words/protocols/jabber/xmlstream.py [hunk 6]
```diff
@@ -715,7 +727,7 @@ def upgradeWithIQResponseTracker(xs):
 
     This makes an L{XmlStream} object provide L{IIQResponseTracker}. When a
     response is an error iq stanza, the deferred has its errback invoked with a
-    failure that holds a L{StanzaException<error.StanzaException>} that is
+    failure that holds a L{error.StanzaError} that is
```
Comment (ralphm) [inline twisted/words/protocols/jabber/xmlstream.py]: Why this change? Is having shorter hot text no longer supported?

File: twisted/words/xish/xmlstream.py [hunk 7]
```diff
@@ -14,6 +14,20 @@
 usable. See Twisted Words for Jabber specific protocol support.
 
 Maintainer: Ralph Meijer
+
+@var STREAM_CONNECTED_EVENT: Token indicating the connection has been
+    established.
+@type STREAM_CONNECTED_EVENT: L{intern}.
+
+@var STREAM_END_EVENT: Token indicating the connection has been closed.
+@type STREAM_CONNECTED_EVENT: L{intern}.
```
Comment (ralphm) [inline twisted/words/xish/xmlstream.py]: This should read `@type STREAM_END_EVENT`.

File: twisted/words/xish/xmlstream.py [hunk 8]
```diff
@@ -14,6 +14,20 @@
...
+
+@var STREAM_ERROR_EVENT: Token indicating a parse error occurred.
+@type STREAM_ERROR_EVENT: L{intern}.
+
+@var STREAM_START_EVENT: Token indicating the parsing of the document has
+    begin.
```
Comment (ralphm) [inline twisted/words/xish/xmlstream.py]: … has _begun_? Also, more precisely, this event signals that the root element of the XML Stream has been received. For XMPP, this would be the `<stream:stream …>` opening tag.

File: twisted/words/xish/xmlstream.py [hunk 9]
```diff
@@ -14,6 +14,20 @@
 usable. See Twisted Words for Jabber specific protocol support.
 
 Maintainer: Ralph Meijer
+
+@var STREAM_CONNECTED_EVENT: Token indicating the connection has been
+    established.
+@type STREAM_CONNECTED_EVENT: L{intern}.
```
Comment (ralphm) [inline twisted/words/xish/xmlstream.py]: There is no type `intern`, really. Just make this into `L{str}` (the native string).

File: twisted/words/protocols/jabber/xmlstream.py (hunk 6 above)
Comment (adiroiban) [inline twisted/words/protocols/jabber/xmlstream.py]: The test is still supported ... for me error.StanzaError is not much better than StanzaError and it reduce the duplication :)

I have used shorted text :)
